File: SEMESTER3/PemrogramanBerorientasiObjek/Week4/TUGAS/Tugas22.java
/*
Program : Tugas 2
Nama    : Prames Ray Lapian
NPM     : 140810210059
Kelas   : A
Tanggal : 24 September  2022
Desc    : pertemuan 4
*/
import java.util.Scanner;

/**
 *
 * @author prame
 */
class convert{
    Scanner sc = new Scanner(System.in);

    //Variable
    private int num;

    //Constructor #1
    public convert(){
        this.num = 0;
    }

    //Constructor 2
    public convert(int num){
        this.num = num;
    }

    //Setter Angka
    public void setAngka(int num){
        this.num = num;
    }

    //Getter Angka
    public int getAngka(){
        return this.num;
    }

    //Input
    public void inputAngka(){
        System.out.print("Masukkan bilangan 1-10 : ");
        this.num = sc.nextInt();
    }

    //IfCase Proses
    public void ifCase(int num){
        String str = "";
        System.out.print("Menggunakan If\t\t: ");
        if(num == 1) 
            str = "Satu";
        else if(num == 2) 
            str = "Dua";
        else if(num == 3) 
            str = "Tiga";
        else if(num == 4) 
            str = "Empat";
        else if(num == 5) 
            str = "Lima";
        else if(num == 6) 
            str = "Enam";
        else if(num == 7) 
            str = "Tujuh";
        else if(num == 8) 
            str = "Delapan";
        else if(num == 9) 
            str = "Sembilan";
        else if(num == 10) 
            str = "Sepuluh";
        else 
            str = "Invalid Number";
        System.out.println(num + " dibaca '" + str + "'");
    }

    //SwitchCase Proses
    public void switchCase(int num){
        String str = "";
        System.out.print("Menggunakan Switch\t: ");
        switch(num) {
            case 1:
                str = "Satu";
                break;
            case 2:
                str = "Dua";
                break;
            case 3:
                str = "Tiga";
                break;
            case 4:
                str = "Empat";
                break;
            case 5:
                str = "Lima";
                break;
            case 6:
                str = "Enam";
                break;
            case 7:
                str = "Tujuh";
                break;
            case 8:
                str = "Delapan";
                break;
            case 9:
                str = "Sembilan";
                break;
            case 10:
                str = "Sepuluh";
                break;
            default:
                str = "Invalid Number";
                break;
        }
        System.out.println(num + " dibaca '" + str + "'");
    }
}

//---------------------------------------------------------------------------------------
public class Tugas22 {
    public static void main(String[] args) {
        Scanner sc= new Scanner(System.in);
        convert angkaUser = new convert(); //Test Constructor #1
        convert angkaSystem = new convert(5); //Test Constructor #2
        
        //System Input : Angka di input langsung didalam codingan
        System.out.println("[Input Sistem]");
        angkaSystem.ifCase(angkaSystem.getAngka());
        angkaSystem.switchCase(angkaSystem.getAngka());
        System.out.println();

        //User Input : Angka di input manual oleh User
        System.out.println("\n[Input Manual]");
        angkaUser.inputAngka();
        angkaUser.ifCase(angkaUser.getAngka());
        angkaUser.switchCase(angkaUser.getAngka());
    
        sc.close();
    }
}

File: SEMESTER3/PemrogramanBerorientasiObjek/Week3/TUGAS/Tugas22.py
 Nama : Amir Salim , Andre Nathaniel Adipraja , Prames Ray lapian , Ibrahim Dafi Iskandar
# NPM : 140810210015 , 140810200042 , 140810210059 , 140810210039
# Kelas : A
# Tanggal : 9 September 2022
# Nama Program : soal2.py
# Deskripsi : Print huruf sesuai angka yang diinput

def inputAngka():
    print("\n--- Input Nilai ---\n")
    hasil = int(input("Masukkan Nilai : "))
    return hasil

def bilIfElse(n):
    print("\n--- Proses Menggunakan If-Else ---\n")
    hasil = " "

    if(n==1):
        hasil="Satu"
    elif(n==2):
        hasil="Dua"
    elif(n==3):
        hasil = "Tiga"
    elif(n==4):
        hasil = "Empat"
    elif(n==5):
        hasil = "Lima"
    elif(n==6):
        hasil = "Enam"
    elif(n==7):
        hasil = "Tujuh"
    elif(n==8):
        hasil = "Delapan"
    elif(n==9):
        hasil = "Sembilan"
    elif(n==10):
        hasil = "Sepuluh"
    else:
        hasil="Invalid Number"
    
    return hasil

def bilMatch(n):
    print("\n--- Proses Menggunakan Match ---\n")
    hasil = " "

    match n:
        case 1:
            hasil="Satu"
        case 2:
            hasil ="Dua"
        case 3:
            hasil="Tiga"
        case 4:
            hasil="Empat"
        case 5:
            hasil="Lima"
        case 6:
            hasil="Enam"
        case 7:
            hasil="Tujuh"
        case 8:
            hasil="Delapan"
        case 9:
            hasil = "Sembilan"
        case 10:
            hasil = "Sepuluh"
        case _:
            hasil = "Invalid Number"
    return hasil

def outputProg (nil, s1,s2):
    print("\n--- Output Program")
    print("Bilangan : " , nil)
    print("Hasil Menggunakan if-else : " , s1)
    print("Hasil Menggunakan Match : " , s2)

nilai = inputAngka()
hasilIfElse=bilIfElse(nilai)
hasilMatch = bilMatch(nilai)

outputProg(nilai,hasilIfElse,hasilMatch)